src/ex15_js-oop/task-01.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+function Candy(brand, weight) {
+  this.brand = brand;
+  this.weight = weight;
+}
+function SugarCandy(brand, weight, type) {
+  Candy.call(this, brand, weight);
+  this.type = type;
+}
+function ChocolateCandy(brand, weight, type) {
+  Candy.call(this, brand, weight);
+  this.type = type;
+}
+SugarCandy.prototype = Object.create(Candy.prototype);
+SugarCandy.prototype.constructor = SugarCandy;
+ChocolateCandy.prototype = Object.create(Candy.prototype);
+ChocolateCandy.prototype.constructor = ChocolateCandy;
+function Gift() {
+  this.candies = [];
+}
+Gift.prototype.addCandy = function(candy, qty) {
+  for (let i = 0; i < qty; i++) {
+    this.candies.push(candy);
+  }
+  return this;
+}
+Gift.prototype.getTotalWeight = function() {
+  let totalWeight = 0;
+  this.candies.forEach(function(candy) { totalWeight += candy.weight });
+  return totalWeight.toFixed(3);
+}
+Gift.prototype.sortCandies = function(byProp, selectedType) {
+  switch(byProp) {
```

Comment:
it will be better if u use `if-else` operator, because u have just 2 conditions
`if (byProps === type) {} else {}`
change that please

ps: `default` need to be in the bottom of the construction

Reply:
I understand that but I kept the scalabilty in mind😎 If there will be more properties to sort wouldn't it be better to use switch? And what if one of the options and the default option are the same should I duplicate them?

Reply:
nope. of course if u have 4 or more conditions it's good idea to use switch
but in this case it's not nessessary
looks like mistake

Reply:
Changed to `if` condition